ios/main.mm

```diff
@@ -15,9 +15,53 @@
 #include "base/NativeApp.h"
 #include "profiler/profiler.h"
 
-#define CS_OPS_STATUS	   0   /* return status */
-#define CS_DEBUGGED		 0x10000000  /* process is currently or has previously been debugged and allowed to run with invalid pages */
+#define CS_OPS_STATUS	0		/* return status */
+#define CS_DEBUGGED	0x10000000	/* process is currently or has previously been debugged and allowed to run with invalid pages */
+#define PT_ATTACHEXC	14		/* attach to running process with signal exception */
+#define PT_DETACH	11		/* stop tracing a process */
 int csops(pid_t pid, unsigned int  ops, void * useraddr, size_t usersize);
+#define ptrace(a, b, c, d) syscall(SYS_ptrace, a, b, c, d)
+
+bool get_debugged() {
+	int flags;
+	int rv = csops(getpid(), CS_OPS_STATUS, &flags, sizeof(flags));
+	if (rv==0 && flags&CS_DEBUGGED) return true;
+
+	pid_t pid = fork();
+	if (pid) {
```

Comment:
This `if` chain is a bit wonky. `if(pid)` basically means `if(pid != 0)`. The `else if(pid == 0)` captures all possible other values for `pid` so the `else` block below contains dead code.

Also, according to [the `fork(2)` man page](https://www.freebsd.org/cgi/man.cgi?fork(2)#RETURN_VALUES) errors return -1. 

So I think this chain should be something like:

```c
if(pid == -1) {
    // fail
} else if(pid == 0) {
    // child process
} else {
    // parent process
}
```

Reply:
Yeah that was supposed to be `pid>0` 😒

Reply:
ah, makes sense lol